raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1004,32 +1003,28 @@ private boolean handleFetchResponse(
 
         FollowerState state = quorum.followerStateOrThrow();
         if (error == Errors.NONE) {
-            FetchResponseData.OffsetAndEpoch nextOffsetAndEpoch = partitionResponse.nextOffsetAndEpoch();
-            if (nextOffsetAndEpoch.nextFetchOffset() > 0) {
+            long truncationOffset = partitionResponse.truncationOffset();
+            if (truncationOffset >= 0) {
```

Comment:
This is the only semantic change in this PR. Couldn't the leader ask us to truncate the entire log? If that is the case wouldn't the leader return a `truncationOffset` of `0`?